catch/unit/memory/hipFreeMipmappedArray.cc

```diff
@@ -0,0 +1,142 @@
+/*
+Copyright (c) 2022 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+
+#include <hip_test_common.hh>
+#include "hipArrayCommon.hh"
+#include "utils.hh"
+
+/*
+ * hipFreeMipmappedArray API test scenarios
+ * 1. Check that hipFreeMipmappedArray implicitly synchronises the device.
+ * 2. Perform multiple allocations and then call hipFreeMipmappedArray on each pointer concurrently (from unique
+ * threads) for different memory types and different allocation sizes.
+ * 3. Pass nullptr as argument and check that correct error code is returned.
+ * 4. Call hipFreeMipmappedArray twice on the same pointer and check that the implementation handles the second
+ * call correctly.
+ */
+
+TEMPLATE_TEST_CASE("Unit_hipFreeMipmappedArrayImplicitSyncArray", "", char, float) {
+#ifdef _WIN32
```

Comment:
Why this?

Reply:
@mangupta @chrispaquot This has been added in line with all other tests that use API that is only functional on Windows OS. This is the case for Mipmapped Array API.

Reply:
same as other comment: use json file instead.

Reply:
@chrispaquot @mangupta On out system, Mipmapped array APIs only build on Linux, for that reason the tests have been disabled using _WIN32

Reply:
I don't understand, the APIs are OS agnostic so should be fine on either ones.

Reply:
@chrispaquot Same as with texture management containing Mipmapped APIs, I can remove the win32 defines, and wait for the CL report to check if everything works properly on Linux on your setup. Would that be a good solution from your side?

Reply:
@mangupta

Reply:
@mangupta win32 defines have been removed, please run the CI to check if everything is working correctly regarding linux/windows test build and execution, and report if there are any bugs because we could not run these tests on our systems.